tests/integration/schema_registry/test_avro_serializers.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import random
+
+import pytest
+
+from confluent_kafka.schema_registry.config import RegistrySerializerConfig
+from confluent_kafka.serialization import AvroDeserializer, AvroSerializer
+from confluent_kafka.serialization import MessageField, SerializationContext
+
+
+@pytest.mark.parametrize("avsc, data",
+                         [('basic_schema.avsc', {'name': 'abc'}),
+                          ('primitive_string.avsc', u'Jämtland'),
+                          ('primitive_bool.avsc', True),
+                          ('primitive_float.avsc', random.uniform(-32768.0, 32768.0)),
+                          ('primitive_double.avsc', random.uniform(-32768.0, 32768.0))])
+def test_avro_record_serialization(kafka_cluster, load_avsc, avsc, data):
+    """
+    Tests basic Avro serializer functionality
+
+    Args:
+        kafka_cluster (KafkaClusterFixture): cluster fixture
+        load_avsc (callable(str)): Avro file reader
+        avsc (str) avsc: Avro schema file
+        data (object): data to be serialized
+
+    Raises:
+        AssertionError on test failure
+
+    """
+    topic = kafka_cluster.create_topic("serialization-avro")
+    schema = load_avsc(avsc)
+    sr_conf = kafka_cluster.schema_registry()
+
+    value_serializer = AvroSerializer(RegistrySerializerConfig(sr_conf),
+                                      schema=load_avsc(avsc))
+
+    value_deserializer = AvroDeserializer(RegistrySerializerConfig(sr_conf),
+                                          schema=load_avsc(avsc))
+
+    producer = kafka_cluster.producer(value_serializer=value_serializer)
+
+    producer.produce(topic, data)
+    producer.flush()
+
+    consumer = kafka_cluster.consumer(value_deserializer=value_deserializer)
+    consumer.subscribe([topic])
```

Comment:
To speed up this test I suggest you produce to partition 0 and then assign consumer to partition 0.